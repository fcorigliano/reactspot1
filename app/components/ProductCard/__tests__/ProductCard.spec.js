const React = require('react');
const ProductList = require('..');
const { render, screen } = require('@testing-library/react');
const product = require('./sample.json');

describe('El componente ProductCard', () => {
    let component;
    const i18n = { gettext: text => text };
    const { id, title, price, thumbnail } = product;

    beforeEach(() => {
        component = render(
            <ProductList 
                i18n={i18n} 
                id={id}
                title={title}
                price={price}
                thumbnail={thumbnail}
            />);
    });

    test('1) Renderiza', () => {
        const { asFragment } = component; 
        expect(asFragment()).toMatchSnapshot();
    });

    test('2) Renderiza un producto con su tÃ­tulo, precio e imagen', () => {
        const title = screen.getByText(/samsung galaxy/i);
        const price = screen.getByText(/38999/i);
        const img = screen.getByRole('img');
        expect(title).toBeInTheDocument();
        expect(price).toBeInTheDocument();
        expect(img).toBeInTheDocument();
    })
});