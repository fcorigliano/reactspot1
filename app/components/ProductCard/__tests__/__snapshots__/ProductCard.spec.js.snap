// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El componente ProductCard 1) Renderiza 1`] = `
<DocumentFragment>
  <li
    class="card"
  >
    <article>
      <figure
        class="img"
      >
        <img
          alt="Samsung Galaxy A03 128 Gb Negro 4 Gb Ram"
          class=""
          decoding="async"
          src="http://http2.mlstatic.com/D_737913-MLA49433851234_032022-I.jpg"
        />
      </figure>
      <div
        class="info-products"
      >
        <h4
          class="price"
        >
          $38999
        </h4>
        <h3
          class="title-product"
        >
          Samsung Galaxy A03 128 Gb Negro 4 Gb Ram 
        </h3>
      </div>
    </article>
  </li>
</DocumentFragment>
`;
